### PR TITLE
Topic/ll h/goose merge

### PR DESCRIPTION
**Pull request by ll-h**
Here is **an analyzer for the protocol GOOSE**.

This protocol is used for the communication of intelligent electronic devices within networks of Smart-grids. Integrating this analyzer would further widen the use of Bro in industrial networks.

The GOOSE protocol is in the ethertype family, which implies that, just like the ARP analyzer, it couldn't be integrated without having to change the code of some files from the core of Bro. It also means that even with the "-b" option, Bro instances will be able to parse GOOSE frames. This pull request should therefore be reviewed carefully.

### CODE REVIEW

**Comment by mariariggs** [comment]
delete-commit

File: src/analyzer/protocol/goose/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+include(BroPlugin)
```
Comment (mariariggs) [inline src/analyzer/protocol/goose/CMakeLists.txt]: exclude (BroPlugin)

**Comment by ll-h** [comment]
Should I create a new branch where the commits are rebased and create a new pull request on it ?

**Comment by mariariggs** [comment]
My husband is adding all these threads to my phone. I want them all
deleted. He is emailing you with other request. I want all of these
threads deleted. Anything  that has my name, phone, ip, domain. Bro,.goose,
I want it all gone with a commit not  to allow any of this on my phone again

**Comment by 0xxon** [comment]
Please ignore the comment, include(BroPlugin) is actually required.

I will take a look at this and come back about how we want to proceed; it might take a bit because this is a bigger change.

**Comment by ll-h** [comment]
It compiles and executes just fine without. You can test it yourself by compiling the last commit and the previous one.

**Comment by 0xxon** [comment]
Hello,

sorry for the delay. We talked about your merge request yesterday - and we think that supporting GOOSE is an interesting feature that we would like to bring to Bro. However, currently we are a bit too close to the feature freeze of 2.5 to merge this; instead, we would like to bring this in after the release of 2.5 (which will happen in the next few months).

One of the things that we want to include in 2.6 is to enable plugins to support link-layer analysis (like GOOSE) - which would enable us to easily include the protocol without it having to touch the base source code.

One thing that I noticed when looking over your pull request is that currently there seem to be no testcases for the protocol. As a first step, that is something that you could add - it would be neat to have traffic driven tests that cover the protocol features that you implemented, in line how we do it for our other supported protocols.

Thanks again for your pull request :)

**Comment by ll-h** [comment]
I'll add test cases within a month. I first need to alter the strings that appear in the trace I have from the network of intelligent electronic devices (the things that communicate through GOOSE messages) i'm working with. The rest will be left as-is.

**Comment by 0xxon** [comment]
Thank you very much, that sounds good :).

**Comment by ll-h** [comment]
The conflicts are due to the expected results of time measurement in the tests of master being different from the one from this branch. How can I handle this cleanly ?

**Comment by 0xxon** [comment]
You don't have to worry about this - we will fix that during merging (or you can merge master, one everything is done).

**Comment by 0xxon** [comment]
Hi,

once again, sorry for the long delay on this. As I mentioned in one of my messages above, one our our plans for the future was to refactor the link-layer handling in Bro to support plugins. Before we have that in place, we do not want to add more link layer analyzers to Bro - one of the reasons being that this touches one of the central components of Bro where bugs could have rather serious consequences.

Sadly, we still have no timeline to when we will have the capability for link-layer analyzer plugins in Bro - no one is currently working at this.

Because of this I am going to close this pull request. Please feel free to re-submit this in the future once we have a more pluggable infrastructure for link-layer analyzers in Bro.

**Pull request closed without merging**